account_payment_order/views/account_invoice_view.xml

```diff
@@ -3,61 +3,53 @@
-        </button>
-        <field name="payment_mode_id" position="after">
-            <field name="payment_order_ok" invisible="1"/>
+    <record id="view_move_form" model="ir.ui.view">
+        <field name="name">account_payment_order.view_move_form</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account_payment_partner.view_move_form"/>
+        <field name="arch" type="xml">
+            <button id="account_invoice_payment_btn" position="after">
+                <!-- For customer refunds:
+                 'Add to Direct Debit Order' will deduct the refund from a customer invoice
+                 We could also need a button 'Add to Payment Order' to reimburse
+                 a customer via wire transfer... but I prefer to keep things
+                 simple ; to do that, the user should manually create a payment order
+                 and select the move lines  -->
+                <button name="create_account_payment_line" type="object"
+                        string="Add to Debit Order"
+                        groups="account_payment_order.group_account_payment"
+                        attrs="{'invisible': ['|', '|',
+                                              ('payment_order_ok', '=', False),
+                                              ('state', '!=', 'open'),
+                                              ('type', 'not in', ('out_invoice', 'out_refund'))
+                                             ]}"/>
+                <button name="create_account_payment_line" type="object"
+                        string="Add to Payment Order"
+                        groups="account_payment_order.group_account_payment"
+                        attrs="{'invisible': ['|', '|',
+                                              ('payment_order_ok', '=', False),
+                                              ('state', '!=', 'open'),
+                                              ('type', 'not in', ('in_invoice', 'in_refund'))
+                                             ]}"/>
+            </button>
+            <field name="payment_mode_id" position="after">
+                <field name="payment_order_ok" invisible="1"/>
+            </field>
+            <field name="ref" position="before">
+                <field name="reference_type" required="1" nolabel="1"
+                       attrs="{'readonly':[('state','!=','draft')],
+                               'invisible': [('type', 'not in', ('out_invoice', 'out_refund'))]}"/>
+            </field>
         </field>
-    </field>
-</record>
-
-
-<act_window id="account_invoice_create_account_payment_line_action"
-    multi="True"
-    key2="client_action_multi"
-    name="Add to Payment/Debit Order"
-    res_model="account.invoice.payment.line.multi"
-    src_model="account.invoice"
-    view_mode="form"
-    target="new"/>
+    </record>
 
+    <!-- TODO: Replace with correct definition -->
```

Comment:
Why this todo?

Reply:
I put it in there to change it to the new Odoo manner to register an action.
So it is laready solved.

Reply:
OK, then please remove it